### PR TITLE
Update grpo_trainer_gsm.py

### PR DESCRIPTION
**Pull request by AbhinayBoda**
This commit fixes a TypeError that occurred because the _get_train_sampler method was expected to receive the `train_dataset` (positional) argument but it was missing from the function signature.

### CODE REVIEW

**Comment by AbhinayBoda** [comment]
Also refer to the following changes made by hf to the trainer.py and update the same in grpo_trainer_gsm.py.
keyword argument: learning rate is added to the params. 
`                        self._maybe_log_save_evaluate(
                            tr_loss,
                            grad_norm,
                            model,
                            trial,
                            epoch,
                            ignore_keys_for_eval,
                            start_time,
                            **learning_rate=learning_rate,**
                        )`
https://github.com/huggingface/transformers/blob/ca0a682796b249788118b2876a2f49cb6ad33694/src/transformers/trainer.py#L2617-L2626
